### PR TITLE
[dv,full_chip,extclk] Add clkmgr_external_clk_src_for_lc test

### PR DESCRIPTION
**Pull request by matutem**
Use lc_ctrl_transition test configuring it with external clock.

Signed-off-by: Guillermo Maturana <maturana@google.com>

### CODE REVIEW

**Comment by matutem** [comment]
Weicai, I did "git mv lc_ctrl_transition_test.c lc_ctrrl_transition_impl.c", but then when I added the tiny new lc_ctrl_transition_test.c it seems git got confused and forgot there was a rename pending...

File: sw/device/tests/sim_dv/meson.build [hunk 1]
```diff
@@ -108,17 +108,45 @@ sw_tests += {
...
-      sw_lib_runtime_log,
       sw_lib_mmio,
       sw_lib_runtime_hart,
+      sw_lib_runtime_log,
+      top_earlgrey,
+    ],
+  ),
+)
+
+clkmgr_external_clk_src_for_lc_test_lib = declare_dependency(
```
Comment (cindychip) [inline sw/device/tests/sim_dv/meson.build:127]: Thanks Matute, could you also add a bazel rule for the test here: https://github.com/lowRISC/opentitan/blob/master/sw/device/tests/BUILD
SW team is in transition to move to bazel.

**Comment by weicaiyang** [comment]
> Weicai, I did "git mv lc_ctrl_transition_test.c lc_ctrrl_transition_impl.c", but then when I added the tiny new lc_ctrl_transition_test.c it seems git got confused and forgot there was a rename pending...

I did this and it work for me.
- git mv A B; 
- create another file A
- git add A

File: sw/device/tests/sim_dv/lc_ctrl_transition_impl.c [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+    LOG_INFO("Waiting for LC transtition done and reboot.");
+    wait_for_interrupt();
+
+  } else {
+    // LC Dev is the requested state from previous lc_state transition request.
+    // Once the sequence checks current state and count via CSRs, the test can
+    // exit successfully.
+    CHECK(curr_state == kDifLcCtrlStateDev, "State transition failed!");
+    check_lc_state_transition_count(kLcStateTransitionCount + 1);
+    return true;
```
Comment (weicaiyang) [inline sw/device/tests/sim_dv/lc_ctrl_transition_impl.c:115]: when use_ext_clk = 1, seems like we don't really know if the clock switch happens or not. If it's still using internal clock, we actually don't know.

Comment (matutem) [reply sw/device/tests/sim_dv/lc_ctrl_transition_impl.c:115]: True. We could configure the external clock frequency to be 96 MHz and confirm by inspection that the test takes less time, or via waves. But it seems there is no other way to confirm this. I'll see if someone comes up with a suggestion from the chat.

**Comment by matutem** [comment]
Weicai, regarding the git operations: I used that sequence and it would seem to work, but it doesn't really do what is intended since the diff does not reflect the move.

**Comment by jadephilipoom** [comment]
> Weicai, regarding the git operations: I used that sequence and it would seem to work, but it doesn't really do what is intended since the diff does not reflect the move.

It should work if the rename and the add are in separate commits, e.g.:
```shell
$ git mv A B
$ git commit
$ git add A
$ git commit
```
With that sequence, the GitHub interface will probably still show the full diff, but reviewers can click on individual commits to see just-a-rename for the first commit and a small diff for the second one.

File: sw/device/tests/sim_dv/meson.build (hunk 1 above)
Comment (matutem) [reply sw/device/tests/sim_dv/meson.build:127]: Done

**Comment by matutem** [comment]
@jadephilipoom @weicaiyang Jade's suggestion should work... It would require me to do some git gymnastics, or create a new PR altogether... The commit with git mv would also require changing meson.build so the test uses the moved file.

I will create a new PR with two commits, so expect this one to disappear.

**Comment by matutem** [comment]
Dear reviewers, this PR is dead: please review https://github.com/lowRISC/opentitan/pull/12225 instead: it contains
two commits so the "git mv" operation survives, and the history of lc_ctrl_transition_test.c is preserved.

**Comment by matutem** [comment]
Wiping this out.

**Pull request closed without merging**